### PR TITLE
Clean up Generator project

### PR DESCRIPTION
**Pull request by jpreese**
- Removes the mapped lib/ directory from Lib2GitSharp
- Removes the user specific config file and adds it to the gitignore

### CODE REVIEW

File: generators/Generators.csproj [hunk 1]
```diff
@@ -15,4 +15,9 @@
   <ItemGroup>
     <EmbeddedResource Include="Output\Templates\*.liquid" />
   </ItemGroup>
+  <ItemGroup>
```
Comment (ErikSchierboom) [inline generators/Generators.csproj]: Why did you add this item group? Was there anything specific going wrong?

Comment (jpreese) [inline generators/Generators.csproj]: Hm.. on my IDE I was getting a big red x next to the lib/ directory as well as seeing the lib2git DLL. You can see my issue here, but unfortunately the opener just closed it without much resolve.

https://github.com/libgit2/libgit2sharp/issues/1425

Comment (ErikSchierboom) [inline generators/Generators.csproj]: @jpreese I'll take a look into this tomorrow

Comment (jpreese) [inline generators/Generators.csproj]: @ErikSchierboom yeah I'm not happy with this. 

I just came back to this and it looks like my red x's are back, even with this change. I'll need to dig into the root cause some more to find out why these are popping up.

Comment (ErikSchierboom) [inline generators/Generators.csproj]: @jpreese Okay, good luck! I haven't encountered those in VS2017.

**Comment by jpreese** [comment]
My gut says this is something related specifically to my machine and not necessarily a fault of the csproj as I originally thought. I'll hold out to see what the Lib2Git guys say. 

In the meantime, I've updated this branch to only include the user settings removal.

**Comment by ErikSchierboom** [comment]
Thanks!

**Comment by jpreese** [comment]
Not sure why I'm a special flower, but the maintainers said they plan on changing how lib/ gets added to projects. So this should be a non issue when they release the newest version.
 (https://github.com/libgit2/libgit2sharp/issues/1425#issuecomment-324350369)

**Comment by ErikSchierboom** [comment]
Awesome! Well done!
